test/lib/ansible_test/_internal/ci/__init__.py

```diff
@@ -99,3 +109,80 @@ def get_ci_provider():  # type: () -> CIProvider
     get_ci_provider.provider = provider
 
     return provider
+
+
+class AuthHelper(ABC):
+    """Public key based authentication helper for Ansible Core CI."""
+    def sign_request(self, request):  # type: (t.Dict[str, t.Any]) -> None
+        """Sign the given auth request and make the public key available."""
+        payload_bytes = json.dumps(request, sort_keys=True).encode()
+        signature_base64_string = self.sign_bytes(payload_bytes)
+
+        request.update(signature=signature_base64_string)
+
+    def initialize_private_key(self):  # type: () -> bytes
+        """
+        Initialize and publish a new key pair (if needed) and return the private key.
+        The private key is cached across ansible-test invocations so it is only generated and published once per CI job.
+        """
+        path = os.path.expanduser('~/.ansible-core-ci-private.key')
```

Comment:
Should this be bytes so we get a byte string back?

Reply:
No, but `to_bytes` should be used on the path being passed to `os.path.exists` -- I'll fix that.

The other functions accepting `path` already handle the conversion.

I have another change in progress that uses an abstraction over the `os` functions to handle all the encoding in one place, so eventually the need for calling `to_bytes` manually for `os` functions will go away.